### PR TITLE
Open and use Core_kernel

### PR DESCRIPTION
**Pull request by jspada**
minor ffmul gadget improvement

### CODE REVIEW

File: src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml [hunk 1]
```diff
@@ -816,7 +818,7 @@ let%test_unit "foreign_field_mul gadget" =
                   (module Runner.Impl)
                   product
               in
-              assert (product = expected) ) ;
+              assert (Stdlib.(product = expected)) ) ;
```
Comment (rbonichon) [inline src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml:821]: Wondering whether the code below is not less error-prone - it probably checks the same thing but you never know ;-)
```suggestion
               let assert_eq ((a, b, c) : field standard_limbs)
                  ((x, y, z) : field standard_limbs) =
                let open Runner.Impl.Field in
                Assert.equal (constant a) (constant x) ;
                Assert.equal (constant b) (constant y) ;
                Assert.equal (constant c) (constant z)
              in
              assert_eq product expected  );
```

File: src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml [hunk 2]
```diff
@@ -883,10 +885,10 @@ let%test_unit "foreign_field_mul gadget" =
                   (module Runner.Impl)
                   product
               in
-              assert (product = expected) ) ;
+              assert Stdlib.(product = expected) ) ;
```
Comment (rbonichon) [inline src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml:888]: Same idea here if the idea is actually correct.

File: src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml [hunk 3]
```diff
@@ -883,10 +885,10 @@ let%test_unit "foreign_field_mul gadget" =
...
 
           (* TODO: 2) Add result bound addition gate *)
-          assert (1 == List.length external_checks.bounds) ;
+          assert (1 = List.length external_checks.bounds) ;
```
Comment (rbonichon) [inline src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml:891]:
```suggestion
          assert (Mina_stdlib.List.Length.equal external_checks.bounds 1) ;
```
List.length runs in O(n) where n is the length of the list. The function above runs in time the length of the cst you pass.

That also means you should add `Mina_stdlib` as dune library dependency.

**Review by rbonichon — COMMENTED** [review]
One take-away: in OCaml never use `List.length` to compare a list length to an integer ;-)

File: src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml (hunk 1 above)
Comment (jspada) [reply src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml:821]: I prefer safety.  Thanks!

**Review by rbonichon — APPROVED** [review]

**Comment by jspada** [comment]
!ci-build-me

**Pull request merged**